src/vtarget-out/vtarget_gen_relchc.cc

```diff
@@ -0,0 +1,394 @@
+%setundef -undriven -init -expose%
+sim -clock clk -reset rst -rstlen %rstlen% -n %cycle% -w %module%
+memory_dff -wr_only
+memory_collect;
+memory_unpack
+splitnets -driver
+opt;;
+memory_collect;
+pmuxtree
+proc
+opt;;
+)***";
+
+
+VlgSglTgtGen_Relchc::VlgSglTgtGen_Relchc(
+    const std::string&
+        output_path, // will be a sub directory of the output_path of its parent
+    const InstrPtr& instr_ptr, // which could be an empty pointer, and it will
+                               // be used to verify invariants
+    const InstrLvlAbsPtr& ila_ptr, const VerilogGenerator::VlgGenConfig& config,
+    nlohmann::json& _rf_vmap, nlohmann::json& _rf_cond, VlgTgtSupplementaryInfo & _sup_info,
+    VerilogInfo* _vlg_info_ptr, const std::string& vlg_mod_inst_name,
+    const std::string& ila_mod_inst_name, const std::string& wrapper_name,
+    const std::vector<std::string>& implementation_srcs,
+    const std::vector<std::string>& implementation_include_path,
+    const vtg_config_t& vtg_config, backend_selector backend,
+    const target_type_t& target_tp,
+    advanced_parameters_t* adv_ptr )
+    : VlgSglTgtGen(output_path, instr_ptr, ila_ptr, config, _rf_vmap, _rf_cond, _sup_info,
+                   _vlg_info_ptr, vlg_mod_inst_name, ila_mod_inst_name,
+                   wrapper_name, implementation_srcs,
+                   implementation_include_path, vtg_config, backend,
+                   target_tp, adv_ptr) { 
+    
+    ILA_ASSERT(! _vtg_config.YosysSmtArrayForRegFile)
+      << "Future work to support array in synthesis";
+    
+    ILA_ASSERT(
+      target_tp == target_type_t::INSTRUCTIONS )
+      << "Only support target type : INSTRUCTIONS, but get: " << target_tp;
+    
+    ILA_ASSERT(! vtg_config.YosysSmtFlattenHierarchy) 
+      << "Monolithic synthesis requires not to flatten hierarchy";
+
+    ILA_ASSERT(! vtg_config.YosysSmtFlattenDatatype)
+      << "BUG: not implemented, future work.";
+
+    }
+
+
+
+/// Add an assumption
+void VlgSglTgtGen_Relchc::add_a_direct_assumption(const std::string& aspt,
+                                                const std::string& dspt) {
+  _problems.assumptions[dspt].exprs.push_back(aspt);
+}
+/// Add an assertion
+void VlgSglTgtGen_Relchc::add_a_direct_assertion(const std::string& asst,
+                                               const std::string& dspt) {
+  _problems.assertions[dspt].exprs.push_back(asst);
+}
+
+/// Pre export work : add assume and asssert to the top level
+/// collect the type of assumptions and assertions
+/// Assertions: variable_map_assert (no invariant_assert --- not for invariant target)
+/// Assumptions:
+///      1. global: noreset, funcmap, absmem, additonal_mapping_control_assume
+///      2. only at starting state:
+///          variable_map_assume, issue_decode, issue_valid, ...
+///      Question: invariant assume? where to put it? Depends on ?
+
+void VlgSglTgtGen_Relchc::PreExportProcess() {
+  // .. ??
+
+  std::string global_assume = "`true";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'global_assume' is assigned a value that is never used.](https://app.codacy.com/manual/Bo-Yuan-Huang/ILAng/pullRequest?prid=4854307)